tests/test_ophys/test_tools_roiextractors.py

```diff
@@ -1054,6 +1055,53 @@ def test_add_fluorescence_traces_regular_timestamps(self):
             self.assertEqual(roi_response_series[series_name].starting_time, times[0])
             self.assertEqual(roi_response_series[series_name].timestamps, None)
 
+    def test_add_fluorescence_traces_irregular_default_timestamps(self):
+        """Test adding traces with default timestamps."""
+
+        segmentation_extractor = generate_dummy_segmentation_extractor(
+            num_rois=2,
+            num_frames=5,
+            num_rows=self.num_rows,
+            num_columns=self.num_columns,
+        )
+
+        add_fluorescence_traces(
+            segmentation_extractor=segmentation_extractor,
+            nwbfile=self.nwbfile,
+            metadata=self.metadata,
+        )
+
+        ophys = get_module(self.nwbfile, "ophys")
+        roi_response_series = ophys.get(self.fluorescence_name).roi_response_series
+        for series_name in roi_response_series.keys():
+            self.assertEqual(roi_response_series[series_name].rate, None)
+            self.assertEqual(roi_response_series[series_name].starting_time, None)
+            assert_array_equal(roi_response_series[series_name].timestamps.data, np.empty(shape=5))
+
+    def test_add_fluorescence_traces_irregular_default_timestamps_with_metadata(self):
+        """Test adding traces with default timestamps and metadata rates."""
+        segmentation_extractor = generate_dummy_segmentation_extractor(
+            num_rois=2,
+            num_frames=5,
+            num_rows=self.num_rows,
+            num_columns=self.num_columns,
+        )
+        metadata = deepcopy(self.metadata)
+        metadata["Ophys"]["Fluorescence"]["roi_response_series"][0].update(rate=1.23)
```

Comment:
Is this second one to test whether we can update / overwrite a specific rate by means of the metadata?

Reply:
This is just showcasing the current issue (which has more to do with how the default behavior adds rate info to the metadata dictionary regardless of if timestamps are set in the segmentation extractor). This test could probably be removed if that default behavior is changed

Reply:
To clarify, default behavior when using any SegmentationInterface (which does stuff to the metadata dictionary where our unit tests of the `add_fluorescence_traces` function does not)